ports/rp2/modules/rp2.py

```diff
@@ -241,10 +241,14 @@ def set(self, dest, data):
 
 
 def asm_pio(**kw):
+    fn = kw.pop("program") if "program" in kw else None
+    args = kw.pop("args") if "args" in kw else {}
```

Comment:
Even simpler, you can use the 2-arg form of `pop` to return the default, eg `fn = kw.pop("program", None)`.